c++/src/kj/async-inl.h

```diff
@@ -719,34 +722,52 @@ class TransformPromiseNode final: public TransformPromiseNodeBase {
         func(kj::fwd<Func>(func)), errorHandler(kj::fwd<ErrorFunc>(errorHandler)) {}
   void destroy() override { freePromise(this); }
 
-  ~TransformPromiseNode() noexcept(false) {
-    // We need to make sure the dependency is deleted before we delete the continuations because it
-    // is a common pattern for the continuations to hold ownership of objects that might be in-use
-    // by the dependency.
-    dropDependency();
-  }
-
 private:
   Func func;
   ErrorFunc errorHandler;
 
   void getImpl(ExceptionOrValue& output) override {
+    typedef _::FixVoid<_::ReturnType<Func, _DepT>> T;
```

Comment:
Deriving T instead of providing it as a template parameter feels clunky now that I look at it again, open to dropping this.

Reply:
Added comments explaining this for now, getting rid of the template parameter appears to contribute to the file size improvement significantly.